.devcontainer/devcontainer.json

```
@@ -1,28 +1,69 @@
 {
   "name": "Jekyll",
   "image": "mcr.microsoft.com/devcontainers/jekyll:2-bullseye",
-  "onCreateCommand": "git config --global --add safe.directory ${containerWorkspaceFolder}",
-  "postCreateCommand": "bash .devcontainer/post-create.sh",
+
+  // Command to set Git configuration upon container creation
+  "onCreateCommand": "git config --global --add safe.directory ${containerWorkspaceFolder} && git config --global user.name 'Your Name' && git config --global user.email 'your.email@example.com'",
+
+  // Command to execute after the container is created
+  "postCreateCommand": "bash .devcontainer/post-create.sh && gem install jekyll bundler",
+
+  // Customizing VSCode settings for a better developer experience
   "customizations": {
     "vscode": {
       "settings": {
-        "terminal.integrated.defaultProfile.linux": "zsh"
+        // Set the default terminal profile to Zsh for better shell experience
+        "terminal.integrated.defaultProfile.linux": "zsh",
+
+        // Enable auto-save of files when the window focus changes
+        "files.autoSave": "onWindowChange",
+
+        // Enable auto-format on save for consistent code formatting
+        "editor.formatOnSave": true,
+
+        // Use Dark theme for the VSCode editor
+        "workbench.colorTheme": "Dark+ (default dark)",
+
+        // Ruby path for proper integration with Ruby-based tools
+        "ruby.rubyPath": "/usr/local/bin/ruby"
       },
```

> I believe these extensions could be a great addition to the config that's been updated. I find these tools helpful to me while writing blogs, from an end user perspective.

If you're considering adding the [previously mentioned changes](https://github.com/cotes2020/jekyll-theme-chirpy/pull/2226#discussion_r1986257355), it might be better to add this config too... Using the public API enabled the `Language-Tool` extension to work properly.

```suggestion

        // Using public API for language tools extension
        "languageToolLinter.serviceType": "public"
      },
```